### PR TITLE
EPMEDU-718: Add Color and Font Provider to Style Engine

### PR DESCRIPTION
**Pull request by diantyn**

### CODE REVIEW

File: Style/Sources/Style/Engine/Providers/Components/StyleFontPreset.swift [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+import Foundation
+
+/// Describes font names used by the UI components.
+///
+/// - Note: Instances of this type are used by ``StyleFontPreset``.
+public protocol StyleFontNamePreset {
+    /// Prepared bold style font name.
+    var bold: String { get }
+    /// Prepared medium style font name.
```
Comment (EugeniyGubar) [inline Style/Sources/Style/Engine/Providers/Components/StyleFontPreset.swift]: These comments are too obvious, looks like an overkill

File: Style/Sources/Style/Engine/Providers/Default/StyleDefaulltFontPresets.swift [hunk 2]
```diff
@@ -0,0 +1,20 @@
+//
+//  StyleDefaulltTypographies.swift
```
Comment (EugeniyGubar) [inline Style/Sources/Style/Engine/Providers/Default/StyleDefaulltFontPresets.swift]: StyleDefaul-l-tTypographies.swift - extra `L`

File: UIComponents/Sources/UIComponents/Onboarding/OnboardingStepView.swift [hunk 3]
```diff
@@ -38,28 +38,28 @@ public final class OnboardingImageStepView: UIView {
         return item
     }()
 
-    private let imageView: UIImageView = {
+    private lazy var imageView: UIImageView = {
```
Comment (EugeniyGubar) [inline UIComponents/Sources/UIComponents/Onboarding/OnboardingStepView.swift:41]: Could you please clarify this change?

**Review by EugeniyGubar — APPROVED** [review]

**Review by igorTimofiejczyk — APPROVED** [review]

File: UIComponents/Sources/UIComponents/Onboarding/OnboardingStepView.swift (hunk 3 above)
Comment (diantyn) [reply UIComponents/Sources/UIComponents/Onboarding/OnboardingStepView.swift:41]: Due to using the designEngine property at the time this property is created

Comment (EugeniyGubar) [reply UIComponents/Sources/UIComponents/Onboarding/OnboardingStepView.swift:41]: Strange, I don't see `designEngine` property usage here

**Pull request merged**